src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1700,6 +1724,7 @@ fn resolution_failure(
                         item(res),
                         variant
                     ),
+                    ResolutionFailure::MalformedGenerics => format!("this link has malformed generics, try adding a '<' or '>'"),
```

Comment:
This looks like:

```
warning: unresolved link to `Vec::<Box<T>::new`
  --> src/test/rustdoc/intra-doc-link-generic-params.rs:10:58
   |
10 | //! And what about something even harder? That would be [`Vec::<Box<T>::new()`].
   |                                                          ^^^^^^^^^^^^^^^^^^^^^ this link has malformed generics, try adding a '<' or '>'
   |
   = note: `#[warn(broken_intra_doc_links)]` on by default
```

An alternative is to use two `span_labels`, which looks like this:

```
warning: unresolved link to `Vec::<Box<T>::new`
  --> src/test/rustdoc/intra-doc-link-generic-params.rs:10:58
   |
10 | //! And what about something even harder? That would be [`Vec::<Box<T>::new()`].
   |                                                          ^^^^^^^^^^^^^^^^^^^^^
   |                                                          |
   |                                                          this link has malformed generics
   |                                                          try adding a '<' or '>'
   |
   = note: `#[warn(broken_intra_doc_links)]` on by default
```

Reply:
Also, I'm not sure if the error message is the clearest. Maybe something like this?

```suggestion
                    ResolutionFailure::MalformedGenerics => format!("this link has malformed generics, try adding a missing '<' or '>'"),
```

Reply:
What's the best way of providing an inline note _and_ an inline help?